api/v1/server/handlers/v1/workflow-runs/get-timings.go

```diff
@@ -0,0 +1,126 @@
+package workflowruns
+
+import (
+	"sort"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/apierrors"
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/gen"
+	"github.com/hatchet-dev/hatchet/pkg/repository/postgres/dbsqlc"
+	"github.com/hatchet-dev/hatchet/pkg/repository/postgres/sqlchelpers"
+	v1 "github.com/hatchet-dev/hatchet/pkg/repository/v1"
+
+	transformers "github.com/hatchet-dev/hatchet/api/v1/server/oas/transformers/v1"
+
+	openapi_types "github.com/oapi-codegen/runtime/types"
+)
+
+func (t *V1WorkflowRunsService) V1WorkflowRunGetTimings(ctx echo.Context, request gen.V1WorkflowRunGetTimingsRequestObject) (gen.V1WorkflowRunGetTimingsResponseObject, error) {
+	tenant := ctx.Get("tenant").(*dbsqlc.Tenant)
+	tenantId := sqlchelpers.UUIDToStr(tenant.ID)
+	rawWorkflowRun := ctx.Get("v1-workflow-run").(*v1.V1WorkflowRunPopulator)
+
+	workflowRun := rawWorkflowRun.WorkflowRun
+
+	var depth int32 = 0
+
+	if request.Params.Depth != nil {
+		depth = int32(*request.Params.Depth)
+	}
+
+	if depth > 10 {
+		return gen.V1WorkflowRunGetTimings400JSONResponse(
+			apierrors.NewAPIErrors("depth must be less than or equal to 10"),
+		), nil
+	}
+
+	taskTimings, idsToDepths, err := t.config.V1.OLAP().GetTaskTimings(
+		ctx.Request().Context(),
+		tenantId,
+		workflowRun.ExternalID,
+		depth,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	result := transformers.ToTaskTimings(taskTimings, idsToDepths)
+
+	// Search for api errors to see how we handle errors in other cases
+	return gen.V1WorkflowRunGetTimings200JSONResponse(
+		gen.V1TaskTimingList{
+			Rows: DFSOrder(result),
+		},
+	), nil
+}
+
+func DFSOrder(tasks []gen.V1TaskTiming) []gen.V1TaskTiming {
+	// build set of task IDs
+	idSet := make(map[openapi_types.UUID]struct{}, len(tasks))
+	for _, t := range tasks {
+		idSet[t.TaskExternalId] = struct{}{}
+	}
+
+	// group children and collect roots
+	children := make(map[openapi_types.UUID][]gen.V1TaskTiming)
+	var roots []gen.V1TaskTiming
```

Comment:
not sure what's idiomatic but I've been making a slice of `V1TaskTiming` of the same length as `tasks` and filling it in in each iter of the loop